zjit/src/codegen.rs

```diff
@@ -859,6 +875,11 @@ fn gen_intern(asm: &mut Assembler, val: Opnd, state: &FrameState) -> Opnd {
     asm_ccall!(asm, rb_str_intern, val)
 }
 
+fn gen_string_length(asm: &mut Assembler, recv: Opnd, state: &FrameState) -> Opnd {
+    gen_prepare_leaf_call_with_gc(asm, state);
+    asm_ccall!(asm, rb_str_length, recv)
```

Comment:
`rb_str_length` returns a `VALUE` -- an `Integer`. you annotated this function as returning a C long but it does not. this means you will be comparing fixnums (shifted and tagged) against non-fixnums. you should probably have a function that wraps `str_strlen` instead. check what YJIT has here

Reply:
yjit doesn't actually have something here so you will have to wrap this and expose `rb_yarv_string_length` in `string.c` and declare it in `zjit.c`